js/bedarf.js

```diff
@@ -1,48 +1,98 @@
 const VMView = {
-  makeCard(vm) {
+  makeCard(type, vm) {
     // was fuer infos haben wir hier? die felder hier sind jz ausgedacht
-    let frag = new DocumentFragment()
-    let info = frag.appendChild(document.createElement('div'))
-    info.appendChild(document.createElement('h3')).textContent = `${vm.ipAdress}`
-    info.appendChild(document.createElement('span')).textContent = `Typ: ${vm.virtualMachineType}`
-    info.appendChild(document.createElement('span')).textContent = `Username/Password:`
-    info.appendChild(document.createElement('span')).textContent = `${vm.username}`
-    info.appendChild(document.createElement('span')).textContent = `${vm.initialPassword}`
-
-    let status = frag.appendChild(document.createElement('div'))
-    let statusText = status.appendChild(document.createElement('h3'))
-    if (vm.status === 'genehmigt') {
-      //string kommt vom backend dann
-      statusText.textContent = 'Genehmigt'
-      statusText.style.color = 'green'
-    } else if (vm.status === 'abgelehnt') {
-      statusText.textContent = 'Abgelehnt'
-      statusText.style.color = 'red'
-    } else {
-      statusText.textContent = 'Offen'
-      statusText.style.color = 'orange'
+
+
+    if (type === 'available'){
+      let frag = new DocumentFragment()
+      let info = frag.appendChild(document.createElement('div'))
+      info.appendChild(document.createElement('h3')).textContent = `${vm.ipAddress}`
+      info.appendChild(document.createElement('span')).textContent = `Typ: ${vm.virtualMachineType}`
+      info.appendChild(document.createElement('span')).textContent = `Username/Password:`
+      info.appendChild(document.createElement('span')).textContent = `${vm.username}`
+      info.appendChild(document.createElement('span')).textContent = `${vm.initialPassword}`
+
+      frag.appendChild(document.createElement('p')).textContent = vm.description
+
+      let card = makeGenericCard(frag)
+      return card
+
     }
-    frag.appendChild(document.createElement('p')).textContent = vm.description
+    if (type === 'request') {
+      let frag = new DocumentFragment()
+      let info = frag.appendChild(document.createElement('div'))
+      info.appendChild(document.createElement('h3')).textContent = `${vm.employeeEmailAddress}`
+      let vmType = ""

```

Comment:
decide vmType string with dictionary instead of whatever you'd call this.
```js
vmTypes = {0: "WindowsServer2022", 1: "Windows11", 2: "Ubuntu2204"}
let vmType = vmTypes[vm.virtualMachineType]
```